src/arithmetic/aggregate_funcs/agg_funcs.c

```diff
@@ -132,7 +132,7 @@ void Aggregate_Free
 
 	SIValue_Free(ctx->result);
 
-	if(ctx->private_data) agg_func->callbacks.free(ctx->private_data);
+	if(ctx->private_data && agg_func->callbacks.free) agg_func->callbacks.free(ctx->private_data);
```

Comment:
aggregation functions must have both `private_data` and a `free` callback, maybe we should replace this with an assertion